File: go/leetcode/905. Sort Array By Parity.go
func sortArrayByParity(A []int) []int {
    pos := 0
    
    for i := 0; i < len(A); i++ {
        if A[i] % 2 == 0 {
            temp := A[pos]
            A[pos] = A[i]
            A[i] = temp
            pos += 1
        }
    }
    
    return A
}

File: javascript/leetcode/905. Sort Array By Parity.js
/**
 * @param {number[]} A
 * @return {number[]}
 */
var sortArrayByParity = function(A) {
  let even = 0;
  let pos = 0;

  for (; even < A.length; even++) {
    if (A[even] % 2 === 0) {
      let temp = A[pos];
      A[pos] = A[even];
      A[even] = temp;
      pos++;
    }
  }

  return A;
};
